pkg/serverless/serverless.go

```diff
@@ -172,6 +172,13 @@ func callInvocationHandler(daemon *daemon.Daemon, arn string, deadlineMs int64,
 func handleInvocation(doneChannel chan bool, daemon *daemon.Daemon, arn string, requestID string) {
 	log.Debug("Received invocation event...")
 	daemon.ExecutionContext.SetFromInvocation(arn, requestID)
+	if daemon.ExecutionContext.GetColdStartTagsForRequestID(requestID).IsColdStart {
+		// Recompute extra tags on first invocation.
+		// Context: they're first processed after extension registration. Re-computing them
+		// fixes any wrong tags, namely that sometimes the process containing the lambda runtime
+		// variable isn't up yet on first tag computation leading to a wrong runtime tag.
+		daemon.ExtraTags.Tags = nil
+	}
```

Comment:
If we just rely on just the global tags computed in `cmd/serverless/main.go`, we'll sometimes have incorrect `runtime` tags. We get the runtime name by looking for the `AWS_EXECUTION_ENV` variable in all running processes, which contains the runtime name.

My theory is that the process running the lambda isn't up yet, resulting in our runtime tag having the value `provided.al2` occasionally (amazon linux 2, incorrectly indicating a custom runtime).

Because of this, we want to re-compute the cold start tags on the first invocation of the lambda sandbox. 

This means there's still the possibility for the runtime to be incorrectly tagged (for example, if a custom metric is submitted and is processed before we can re-compute these cold start tags), though this is better than the previous behavior (where in the case of this race, no metadata tags would be added at all).